### PR TITLE
fix(core): annotation logic for static pods

### LINKED ISSUE #1635: bug: AppArmor annotation logic for orphan/static pods fails
## Bug Report

**Description**
Kubernetes allows creation of bare/orphan/static pods which don't have an owner. It's not possible for KubeArmor to patch these but the controller can because it has a mutating admission webhook for pods. Currently KubeArmor either panics or tries to annotate a deployment which doesn't even exist while trying to annotate these.

**General Information**

- Environment description: k3s or any K8s
- Kernel version (run `uname -a`): N/A
- Orchestration system version in use (e.g. `kubectl version`, ...): 1.23.9+k3s+1
- Link to relevant artifacts (policies, deployments scripts, ...)
- Target containers/pods

**To Reproduce**

Two scenarios which might have different results:

Scenario 1: pod deployed before KubeArmor
1. Create an orphan pod.
2. Deploy KubeArmor with apparmor enforcer and checkout KubeArmor logs. You'll either see an error while trying to annotate the pod or KubeArmor will panic.

Scenario 2: pod deployed after KubeArmor
1. Deploy KubeArmor with apparmor enforcer
2. Create an orphan pod. It might be annotated by the controller but you'll still see an error log in KubeArmor.

You can create orphan pods with:
```
kubectl run nginx-pod --rm -it --image=nginx
```

**Expected behavior**

KubeArmor should not try to patch orphan pods.

### PR DESCRIPTION
**Pull request by Shreyas220**
**Purpose of PR?**:

Fixes #1635 

**Does this PR introduce a breaking change?**
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

1: pod deployed before KubeArmor
2: pod deployed after KubeArmor

**Additional information for reviewer?** :

kubearmor cannot annotate static pod with apparmor annotations

```
2024-02-19 05:40:04.024021	ERROR	Failed to update AppArmor Annotations (default/nginx-pod/nginx-pod, Pod "nginx-pod" is invalid: metadata.annotations[container.apparmor.security.beta.kubernetes.io/nginx-pod]: Forbidden: may not add AppArmor annotations)
``` 

i still have the logic for annotating static pod in k8shandler , will/not remove based on review 

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/core/k8sHandler.go [hunk 1]
```diff
@@ -287,15 +287,31 @@ func (kh *K8sHandler) PatchResourceWithAppArmorAnnotations(namespaceName, deploy
 		return nil
 
 	} else if kind == "Pod" {
-		_, err := kh.K8sClient.CoreV1().Pods(namespaceName).Patch(context.Background(), deploymentName, types.MergePatchType, []byte(spec), metav1.PatchOptions{})
+		// this condition wont be triggered but keeping here for reference
+
+		pod, err := kh.K8sClient.CoreV1().Pods(namespaceName).Get(context.TODO(), deploymentName, metav1.GetOptions{})
```
Comment (daemon1024) [inline KubeArmor/core/k8sHandler.go]: Why are we doing a Get here?

Comment (Shreyas220) [reply KubeArmor/core/k8sHandler.go]: cannot patch pods 

so here we are getting the pod 
adding annotation and then updating the pods

Comment (Shreyas220) [reply KubeArmor/core/k8sHandler.go]: this is not triggered 
should i just remove this ?

**Review by daemon1024 — APPROVED** [review]

**Review by daemon1024 — CHANGES_REQUESTED** [review]
LGTM 👌🏽 

Please handle fmt/lint errors, and squash commits, rebase to latest.

**Review by daemon1024 — APPROVED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Thanks @Shreyas220!

**Pull request merged**